src/lib.rs

```diff
@@ -192,7 +192,7 @@ pub fn narrowable_libgc(args: TokenStream, input: TokenStream) -> TokenStream {
 
         impl #struct_id {
             /// Create a new narrow pointer to `U: #trait_id`.
-            pub fn new<U>(obj: U) -> ::libgc::Gc<Self>
+            pub fn new<U: Send>(obj: U) -> ::libgc::Gc<Self>
```

Comment:
I'm not saying I object to this, but isn't this saying that `T` *must* be `Send`? The PR description made it sound like it would be a sort-of smart pointer thingy where `#struct_id` is `Send` if `T:Send` but is not `Send` if `!T:Send`?

Reply:
Yeah that was a bit poorly worded actually. `T` _must_ be `Send` in this case, because `NarrowT` is being put in the `Gc` in its place. This doesn't happen by default though because the inner `*mut u8` ptr makes `NarrowT` !Send`.

Reply:
Aha, right, that makes sense!